cmd/vicadmin/server.go

```diff
@@ -53,20 +53,27 @@ func (s *server) listen(useTLS bool) error {
 
 	// FIXME: assignment copies lock value to tlsConfig: crypto/tls.Config contains sync.Once contains sync.Mutex
 	tlsconfig := func(c *tls.Config) *tls.Config {
-		if c.ClientCAs == nil {
-			c.ClientCAs = x509.NewCertPool()
-		}
-		if !c.ClientCAs.AppendCertsFromPEM(vchConfig.CertificateAuthorities) {
-			log.Warnf("Unable to load CAs from config; client auth via certificate will not function")
+		// if there are CAs, then TLS is enabled
+		if len(vchConfig.CertificateAuthorities) != 0 {
+			if c.ClientCAs == nil {
+				c.ClientCAs = x509.NewCertPool()
+			}
+			if !c.ClientCAs.AppendCertsFromPEM(vchConfig.CertificateAuthorities) {
+				log.Errorf("Unable to load CAs from config; client auth via certificate will not function")
+			}
+			c.ClientAuth = tls.RequireAndVerifyClientCert
```

Comment:
This will become `tls.VerifyClientCertIfGiven` when username/password authentication is in place, but making this more permissive in the absence of u/p auth requires bigger changes that won't really make a lot of sense until u/p auth goes in.